pkg/northbound/e2/e2.go

```diff
@@ -18,27 +18,53 @@ import (
 	liblog "github.com/onosproject/onos-lib-go/pkg/logging"
 	service "github.com/onosproject/onos-lib-go/pkg/northbound"
 	"github.com/onosproject/ran-simulator/api/e2"
+	"github.com/onosproject/ran-simulator/api/types"
+	"github.com/onosproject/ran-simulator/pkg/utils"
 	"google.golang.org/grpc"
 )
 
 var log = liblog.GetLogger("northbound", "e2")
 
-// NewService returns a new trafficsim Service
-func NewService() (service.Service, error) {
-	return &Service{}, nil
+// NewTowerServer - start a new gRPC server per tower
+func NewTowerServer(towerIndex int, serverParams utils.ServerParams) error {
```

Comment:
The idea is to have one separate connection from onos-ric to each base station? I mean there is no proxy thing involved correct?

Reply:
that is correct - each base station is listed with its own endpoint address in onos-topo

Reply:
I've tested it up to 1024 base station and have not seen any problems

Reply:
OK. Good. So for now, we deal with 1:1 connections.